### PR TITLE
Page when we are out of capacity

### PR DESCRIPTION
**Pull request by byucesoy**
**Add page type identifier to the list of tags**
Two different types of pages can have same tags, but they should not interfere
with each other. Because of that, we started to add type of the page as one of
the tags we use for deduplication.

**Create a page if no capacity is left**
Ideally we would create a page when capacity is low, but the purpose of this
commit it fixing the blind spot we added in https://github.com/ubicloud/ubicloud/commit/cfa245c3b3e82657558fec08e4d3048669230db5, not having full blown capacity
monitoring system. In that commit, to prevent page storms, we stopped generating
one page per each unsuccessful allocation. With this commit, we create one page
for all failed allocations that are due to lack of capacity. We do this by deduplicating
pages based on tags "NoCapacity" and the location of the requested VM.

### CODE REVIEW

**Comment by fdr** [comment]
What does the person paged do?

**Comment by byucesoy** [comment]
Depending on the situation;
- Communicate with users about potential delays
- Create new hosts
- Ignore

Basically, whatever we are doing already when we get allocation errors due to insufficient capacity. I guess your point is that the page is not actionable, which is a valid argument. The purpose of this PR was bringing back the pages we disabled in https://github.com/ubicloud/ubicloud/commit/cfa245c3b3e82657558fec08e4d3048669230db5, with one notable difference; instead of getting 100s of pages (separately for each VM) we will get one (per region).

File: prog/vm/nexus.rb [hunk 1]
```diff
@@ -196,8 +196,20 @@ def before_run
...
+    vm_host_id = begin
+      allocate
+    rescue RuntimeError => ex
+      raise unless ex.message.include?("no space left on any eligible hosts")
+
+      Prog::PageNexus.assemble("No capacity left at #{vm.location}", "NoCapacity", vm.location)
+
+      Clog.emit("No capacity left") do
+        {lack_of_capacity: {location: vm.location,
+                            queue_size: Vm.join(:strand, id: :id).where(:location => "hetzner-fsn1", Sequel[:strand][:label] => "wait").count}}
```
Comment (enescakir) [inline prog/vm/nexus.rb:208]:
```suggestion
                            queue_size: Vm.join(:strand, id: :id).where(:location ⇨ vm.location, Sequel[:strand][:label] => "wait").count}}
```

**Comment by enescakir** [comment]
> What does the person paged do?

No action is taken for a single capacity page. The oncall can only monitor the issue and ensure we return to normal within a reasonable time frame, such as 30 minutes. If the issue persists for more than 30 minutes, some customers might be keeping all VM hosts busy, preventing other customers from creating runners. The oncall may investigate whether this spike is expected.

If we encounter this page too frequently, it indicates we have a capacity utilization issue. The oncall may then coordinate the purchase of new bare metal servers.

**Review by enescakir — APPROVED** [review]

**Comment by byucesoy** [comment]
Merging this now as this does not increase the total number of pages we would get compared to what we used to get before https://github.com/ubicloud/ubicloud/commit/cfa245c3b3e82657558fec08e4d3048669230db5. If we find the pages non-actionable, we can remove the line that creates pages.

**Pull request merged**